kore/src/Kore/Step/Simplification/Exists.hs

```diff
@@ -245,8 +245,9 @@ matchesToVariableSubstitution
   , Substitution.null boundSubstitution
   , not (TermLike.hasFreeVariable (inject $ variableName variable) term)
   = do
-    matchResult <- matchIncremental first second
-    case matchResult of
+    matchResultFS <- matchIncremental first second
+    matchResultSF <- matchIncremental second first
+    case matchResultFS <|> matchResultSF of
```

Comment:
Why is this necessary?

Reply:
Without it, the `actual` result from [this test](https://github.com/kframework/kore/blob/master/kore/test/Test/Kore/Step/Simplification/Exists.hs#L283) would be
```
\and{testSort{}}(
    /* term: */
    /* Created: Test.Kore.Step.Simplification.Exists.f */
    /* Fn Spa */
    f{}(/* Fl Fn D Sfa Cl */ a{}()),
\and{testSort{}}(
    /* predicate: */
    /* Sfa */
    \exists{testSort{}}(
        x:testSort{},
        /* Sfa */
        \equals{testSort{}, testSort{}}(
            /* Fn Sfa */ f{}(/* Fl Fn D Sfa Cl */ a{}()),
            /* Fn Sfa */ f{}(/* Fl Fn D Sfa */ x:testSort{})
        )
    ),
    /* substitution: */
    /* Spa */
    \equals{testSort{}, testSort{}}(
        /* Fl Fn D Sfa */ y:testSort{},
        /* Fn Sfa */ f{}(/* Fl Fn D Sfa Cl */ a{}())
    )
))
```
instead of having the expected `\top` predicate.

Reply:
This predicate can't exist at this point, because it's not fully simplified:

```
        \equals{testSort{}, testSort{}}(
            /* Fn Sfa */ f{}(/* Fl Fn D Sfa Cl */ a{}()),
            /* Fn Sfa */ f{}(/* Fl Fn D Sfa */ x:testSort{})
        )
```

So, we will leave this as-is for the purposes of this pull request, but please add a `TODO` to this function (`matchesToVariableSubstitution`) saying that it must go away.